midterm-2023-11-06/code/5163-mihaela-milosevska/script.ts

```diff
@@ -0,0 +1,82 @@
+interface album{
+    name: string;
+    year; number;
+}
+
+interface band {
+    id: number;
+    name: string;
+    formed: number;
+    location: string;
+    genre: string;
+    members: string;
+    albums: number;
+}
+type BandSorter = (first: Band, second: Band) => number;
+
+async function getBands(): Promise<Band[]> {
+    const url = "https://raw.githubusercontent.com/sweko/uacs-internet-programming-exams/main/midterm-2023-11-06/data/bands.json";
+    const response = await axios.get(url);
+    return response.data.metalBands;
+}
+
+function formatBand(band: Band): Record<string, unknown> {
+    const members = band.members.sort().slice(0, 5).join(', ');
+    const lastMember = band.members.length > 5 ? '...' : '';
+    const membersString = band.members.length < 3 ? members : `${members}, ${lastMember}`;
+    const [city, country] = band.location.split(', ');

```

Comment:
This only works for two part names. For `London, England` it will correctly produce `England`. However, for `Tampa, Florida, USA` it will produce `Florida` instead of `USA`. Additionally, we're not using the country :D